maven-core/src/main/java/org/apache/maven/artifact/handler/manager/LegacyArtifactHandlerManager.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.maven.artifact.handler.manager;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.apache.maven.artifact.handler.ArtifactHandler;
+import org.apache.maven.artifact.handler.DefaultArtifactHandler;
+
+/**
+ */
+@Named
+@Singleton
+public class LegacyArtifactHandlerManager implements ArtifactHandlerManager {
```

Comment:
Why do we need the `LegacyArtifactHandlerManager` _and_ the `DefaultArtifactHandlerManager` at the same time ?

Reply:
This is a simple trick:
* legacyAHM is what "default" was so far (it have legacy AHs injected), is new, but nobody uses it as it is new thing (just new code "knows" about it)
* defaultAHM changed to delegate calls to typeRegistry
* typeRegistry uses as a fallback legacyAHM

This completes the circle:
* legacy code will refer to defaultAHM that goes to typeRegistry that will refer legacyAHM (as fallback)
* new code will refer to typeRegistry

This way new and legacy code both "see" the same thing: whether it is new AH (legacy) or Type (new) being added by extension for example